metascript/animation.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+
+import subprocess
+import os
+from datetime import datetime
+from math import floor
+
+PERFORMANCE_KEYWORD = "performance:"
+
+def _generate_animation_recorder_vrml(duration, output, controller_name):
+    return (
+        f'DEF ANIMATION_RECORDER_SUPERVISOR Robot {{\n'
+        f'  name "animation_recorder_supervisor"\n'
+        f'  controller "animator"\n'
+        f'  controllerArgs [\n'
```

Comment:
This doesn't seem to be PEP8 compliant as you should get a warning here for using `f` without any `{}` argument.

Reply:
I reused Thomas' code, the f are there to keep everything aligned nicely. I could use a triple quote f-string to wrap the whole vrml code snippet:
```python
def _generate_animation_recorder_vrml(duration, output, controller_name):
    return f'''
    DEF ANIMATION_RECORDER_SUPERVISOR Robot {{
    name "animation_recorder_supervisor"
    controller "animator"
    controllerArgs [
        "--duration={duration}"
        "--output={output}"
        "--controller={controller_name}"
    ]
    supervisor TRUE
    }}
    '''
```
There will be extra indentation in the world file but Webots doesn't seem to worry too much:
(screenshot of the temporary world file modification to add the animator supervisor)
![Screenshot from 2022-11-17 15-56-52](https://user-images.githubusercontent.com/62928319/202480766-ceca0f4e-1677-4578-9570-835e7e242ca7.png)